test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests.cs

```diff
@@ -0,0 +1,209 @@
+{
+    public class FunctionMetadataProviderGeneratorTests
+    {
+        private Assembly[] referencedExtensionAssemblies;
+
+        public FunctionMetadataProviderGeneratorTests()
+        {
+            // load all extensions used in tests (match extensions tested on E2E app? Or include ALL extensions?)
+            var abstractionsExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Abstractions.dll");
+            var httpExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Http.dll");
+            var storageExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Storage.dll");
+            var cosmosDBExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.CosmosDB.dll");
+            var timerExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Timer.dll");
+            var eventHubsExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.EventHubs.dll");
+            var blobExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs.dll");
+            var queueExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Storage.Queues.dll");
+            var loggerExtension = Assembly.LoadFrom("Microsoft.Extensions.Logging.Abstractions.dll");
+            var hostingExtension = Assembly.LoadFrom("Microsoft.Extensions.Hosting.dll");
+            var diExtension = Assembly.LoadFrom("Microsoft.Extensions.DependencyInjection.dll");
+            var hostingAbExtension = Assembly.LoadFrom("Microsoft.Extensions.Hosting.Abstractions.dll");
+            var diAbExtension = Assembly.LoadFrom("Microsoft.Extensions.DependencyInjection.Abstractions.dll");
+
+            referencedExtensionAssemblies = new[]
+            {
+                abstractionsExtension,
+                httpExtension,
+                storageExtension,
+                cosmosDBExtension,
+                timerExtension,
+                eventHubsExtension,
+                blobExtension,
+                queueExtension,
+                loggerExtension,
+                hostingExtension,
+                hostingAbExtension,
+                diExtension,
+                diAbExtension
+            };
+
+        }
+
+        [Fact]
+        public async Task GenerateSimpleHttpTriggerMetadata()
+        {
+            // test generating function metadata for a simple HttpTrigger
+            string inputCode = @"
+            using System;
+            using System.Collections.Generic;
+            using System.Diagnostics;
+            using System.Net;
+            using Microsoft.Azure.Functions.Worker;
+            using Microsoft.Azure.Functions.Worker.Http;
+            using Microsoft.Extensions.Logging;
+
+            namespace FunctionApp
+            {
+                public static class HttpTriggerSimple
+                {
+                    [Function(nameof(HttpTriggerSimple))]
```

Comment:
Consider adding a non function method to this class to make sure everything still works as expected.